### PR TITLE
Update Melos and CD/CI

### LINKED ISSUE #220: integration_test GitHub Actions job broken.
For example: https://github.com/fluttercommunity/plus_plugins/runs/2389416418

### LINKED ISSUE #488: [all]] Transition of platform channel test interfaces to flutter_test package
We need to take care of this change: https://flutter.dev/docs/release/breaking-changes/mock-platform-channels

### PR DESCRIPTION
**Pull request by mhadaily**
## Description
Addressing a few issues and add more jobs in our CD/CI

## Related Issues

fixes #488

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by juliansteenbakker** [comment]
[FYI, you can mark a PR as a draft so it can't be merged yet 😉](https://github.blog/2019-02-14-introducing-draft-pull-requests/)

**Comment by mhadaily** [comment]
> [FYI, you can mark a PR as a draft so it can't be merged yet 😉](https://github.blog/2019-02-14-introducing-draft-pull-requests/)

haha, I know, but it was ready to merge actually.

File: .github/workflows/scripts/install-tools.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
```
Comment (Elias8) [inline .github/workflows/scripts/install-tools.sh:2]: I think we should also globally activate `flutter_plugin_tools` here.

**Comment by juliansteenbakker** [comment]
Will this also update the CI street to flutter 2.5(.2)? Right now #442 is failing because i've already migrated some tests to 2.5.2.

**Comment by mhadaily** [comment]
Yes, it will 

however, I have not updated tests in `main` branch yet.

**Comment by mhadaily** [comment]
@Elias8 let me know if this sounds good to you. 

There are a few things we need to fix later 
1- fixing integration tests
2- fixing build on Linux and windows 
3- add running tests on all three platforms.

**Review by Elias8 — APPROVED** [review]
LGTM! 👍

**Comment by Elias8** [comment]
> There are a few things we need to fix later
1- fixing integration tests
2- fixing build on Linux and windows
3- add running tests on all three platforms.

Sounds good @mhadaily

**Comment by juliansteenbakker** [comment]
This PR added `google-java-format` formatting, however this is not default in android studio. Maybe it's a good idea to add a section to the readme explaining on how to format files correctly since it's not really straight forward.

**Pull request merged**